### PR TITLE
[TH] removed redundant in-code timeout

### PR DESCRIPTION
**Pull request by tank157**
@cap10morgan can you look at this? Ideally the closed-channel-fn will live somewhere common at some point, any suggestions? kehaar.test-helpers maybe?

### CODE REVIEW

File: test/voter_registration_works/reg_form_test.clj [hunk 1]
```diff
@@ -112,3 +113,19 @@
                                 [:context :citizen_no])))
            (is (= nvrf-form-message-optional-fields
                   (registration->nvrf-form registration-optional-fields)))))
+
+(defn closed-channel-fn
+  "Returns a function that returns closed channel, simulating a timeout"
+  []
+  (fn [_]
+    (let [chan (async/chan)
+          _ (async/close! chan)]
```
Comment (cap10morgan) [inline test/voter_registration_works/reg_form_test.clj]: This can be pulled out of the `let` form.

**Comment by cap10morgan** [comment]
I'd be curious to get @tie-rack's thoughts on this.

Chris: We're trying to test a kehaar-backed response channel timing out, but currently to get kehaar to actually do that itself requires wiring up all the RabbitMQ stuff too.

**Comment by tank157** [comment]
I think I'm ok with it...I don't need to test the *actual* timeout, just what my code does when a timeout occurs, in which case we get a nil value when taking from the (closed) channel. So the redef is fine for simulating this.

**Comment by cap10morgan** [comment]
@tank157 Yeah, I'm not necessarily arguing against that. I'm likely fine w/ this approach too. Kehaar closing the channel on timeout is a nice API that we can easily test as you've done here. I'm just curious what Chris thinks about it.

**Comment by tank157** [comment]
Hmmmm, _are_ there any tests in kehaar on the timing out? I didn't see any in the wire_up_test ns.

**Comment by tank157** [comment]
Ok, I've pushed a PR to kehaar to test the timeout functionality of `async->fn` in conjunction with an `external-service`. https://github.com/democracyworks/kehaar/pull/18

**Comment by tie-rack** [comment]
I feel good about this (especially with the attending test on Kehaar that promises that this is what it does). :+1:

**Comment by cap10morgan** [comment]
OK, LGTM too.
